src/v/storage/readers_cache.cc

```diff
@@ -16,6 +16,7 @@
 #include "ssx/future-util.h"
```

Comment:
> Reader cache eviction is am important behavior for performance: if
eviction isn't working as expected we can get a large amount of
additional reads from disk as ss::input_stream read-ahead is dropped
without being used.

Trying to understand this. Is the idea that eviction enables something like a `seek` to the correct location, where as if the eviction doesn't happen we have to read/scan up to the target position?

Reply:
@dotnwat - I wrote this in a confusing way, I'm going to reword it to remove "eviction". I didn't mean that eviction is good for performance, just that the overall operation of the readers cache is important for performance, that is, that "hits" are good for performance.

Hits are good because they let you re-use the readahead.

Reply:
(this has been re-worded)